docs/resources/client/particles.md

```diff
@@ -0,0 +1,254 @@
+    // ParticleProvider, see below. You may also add additional parameters as needed, e.g. xSpeed/ySpeed/zSpeed.
+    public MyParticle(ClientLevel level, double x, double y, double z, SpriteSet spriteSet) {
+        super(level, x, y, z);
+        this.spriteSet = spriteSet;
+        this.gravity = 0; // Our particle floats in midair now, because why not.
+    }
+    
+    @Override
+    public void tick() {
+        // Set the sprite for the current particle age, i.e. advance the animation.
+        setSpriteFromAge(spriteSet);
+        // Let super handle further movement. You may replace this with your own movement if needed.
+        // You may also override move() if you only want to modify the built-in movement.
+        super.tick();
+    }
+}
+```
+
+### `ParticleProvider`
+
+Next, particle types must register a `ParticleProvider`. `ParticleProvider` is a client-only class responsible for actually creating our `Particle`s through the `createParticle` method. While more elaborate code can be included here, many particle providers are as simple as this:
+
+```java
+// The generic type of ParticleProvider must match the type of the particle type this provider is for.
+public class MyParticleProvider implements ParticleProvider<SimpleParticleType> {
+    // A set of particle sprites.
+    private final SpriteSet spriteSet;
+    
+    // The registration function passes a SpriteSet, so we accept that and store it for further use.
+    public MyParticleProvider(SpriteSet spriteSet) {
+        this.spriteSet = spriteSet;
+    }
+    
+    // This is where the magic happens. We return a new particle each time this method is called!
+    // The type of the first parameter matches the generic type passed to the super interface.
+    @Override
+    public Particle createParticle(SimpleParticleType type, ClientLevel level,
+            double x, double y, double z, double xSpeed, double ySpeed, double zSpeed) {
+        // We don't use the type and speed, and pass in everything else. You may of course use them if needed.
+        return new MyParticle(level, x, y, z, spriteSet);
+    }
+}
+```
+
+Your particle provider must then be associated with the particle type in the [client-side][side] [mod bus][modbus] [event] `RegisterParticleProvidersEvent`:
+
+```java
+@SubscribeEvent
+public static void registerParticleProviders(RegisterParticleProvidersEvent event) {
+    // There are multiple ways to register providers, all differing in the functional type they provide in the
+    // second parameter. For example, #registerSpriteSet represents a Function<SpriteSet, ParticleProvider<?>>:
+    event.registerSpriteSet(MyParticleTypes.MY_PARTICLE.get(), MyParticleProvider::new);
+    // Other methods include #registerSprite, which is essentially a Supplier<TextureSheetParticle>,
+    // and #registerSpecial, which maps to a Supplier<Particle>. See the source code of the event for further info.
+}
+```
+
+### Particle Definitions
```

Comment:
This only applies if the sprite set method is used, when the `ParticleRenderType` uses the particles texture atlas.

Reply:
Mind elaborating?

Reply:
Particle descriptions are essentially a list of textures to be stitched into the particle texture atlas. When registering the particle factory, you have to use the sprite particle registration interface to tell the engine that you have textures to be stitched. An error is thrown if there is a description but incorrect factory registry, or vice versa. As the descriptions are stitched into the atlas, only particle render types that use that atlas will actually render the particle correctly.